Add docs related to KEP #950 (startupProbe) (#15714)
kubernetes/website: content/en/docs/concepts/workloads/pods/pod-lifecycle.md

The Pod remains in the unready state while it waits for the Containers in the Pod
to stop.

### When should you use a startup probe?

> {{< feature-state for_k8s_version="v1.15" state="alpha" >}}

Reviewer: ```suggestion
{{< feature-state for_k8s_version="v1.16" state="alpha" >}}
```

> If your Container usually starts in more than `initialDelaySeconds + failureThreshold * periodSeconds` (which equals to 30s with the default values) you should specify a startup probe that checks the same endpoint as the liveness probe.

Reviewer: ```suggestion
 If your Container usually starts in more than `initialDelaySeconds + failureThreshold × periodSeconds` (which quals to 30s with the default values) you should specify a startup probe that checks the same endpoint as the liveness probe.
```

> You should then set its `failureThreshold` high enough to allow the Container to start, without changing the default values of the liveness probe to ensure proper deadlock protection.

Reviewer: ```suggestion
You should then set its `failureThreshold` high enough to allow the Container to start, without changing the default values of the liveness probe. This helps to protect against deadlocks.
```
